packages/components/time-field/src/time-field.ts

```diff
@@ -0,0 +1,610 @@
+import { localized, msg, str } from '@lit/localize';
+import { type ScopedElementsMap, ScopedElementsMixin } from '@open-wc/scoped-elements/lit-element.js';
+import { FormControlMixin, type SlFormControlEvent, type SlUpdateStateEvent } from '@sl-design-system/form';
+import { Icon } from '@sl-design-system/icon';
+import { type EventEmitter, anchor, event } from '@sl-design-system/shared';
+import { type SlBlurEvent, type SlChangeEvent, type SlFocusEvent } from '@sl-design-system/shared/events.js';
+import { FieldButton, TextField } from '@sl-design-system/text-field';
+import { type CSSResultGroup, LitElement, type PropertyValues, type TemplateResult, html } from 'lit';
+import { property, query } from 'lit/decorators.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import styles from './time-field.scss.js';
+
+declare global {
+  interface HTMLElementTagNameMap {
+    'sl-time-field': TimeField;
+  }
+}
+
+/**
+ * A time field control for selecting a time.
+ */
+@localized()
+export class TimeField extends FormControlMixin(ScopedElementsMixin(LitElement)) {
+  /** The default step between each hour option. */
+  static hourStep = 1;
+
+  /** The default step between each minute option. */
+  static minuteStep = 5;
+
+  /** @internal The default offset of the popover to the text-field. */
+  static offset = 6;
+
+  /** @internal */
+  static get scopedElements(): ScopedElementsMap {
+    return {
+      'sl-field-button': FieldButton,
+      'sl-icon': Icon,
+      'sl-text-field': TextField
+    };
+  }
+
+  /** @internal */
+  static override styles: CSSResultGroup = styles;
+
+  /** @internal The default margin between the popover and the viewport. */
+  static viewportMargin = 8;
+
+  /**
+   * Flag indicating whether the popover was just closed. We need to know this so we can
+   * properly handle button clicks that close the popover. If the popover was just closed,
+   * we don't want to show it again when the button click event fires.
+   */
+  #popoverJustClosed = false;
+
+  /** The start time. */
```

Comment:
```suggestion
  /** The start time; the time that has the initial focus when the picker is opened when there isn't a value. */
```

The term `startTime` might be confusing because you will likely use the time-picker to let the user set the start time of something.